server/services/v1/auth_provider.go

```diff
@@ -299,7 +327,59 @@ func getAccessTokenUsingRefreshToken(req *api.GetAccessTokenRequest, a *Auth0) (
 	return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = %s", bodyStr)
 }
 
-func getAccessTokenUsingClientCredentials(req *api.GetAccessTokenRequest, a *Auth0) (*api.GetAccessTokenResponse, error) {
+type tokenMetadataEntry struct {
+	AccessToken string
+	ExpireAt    int64 // unix second
+}
+
+func getAccessTokenUsingClientCredentials(ctx context.Context, req *api.GetAccessTokenRequest, a *Auth0) (*api.GetAccessTokenResponse, error) {
+	// lookup the internal namespace
+	tx, err := a.txMgr.StartTx(ctx)
+	if err != nil {
+		log.Warn().Err(err).Msg("Could not start tx for internal token cache lookup")
+		return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = could not start tx for internal lookup")
```

Comment:
Maybe define a helper: `tokenError("could not start tx for internal lookup")`

Reply:
created helper and deduplicated code.